bionic/persistence.py

```diff
@@ -149,27 +149,50 @@ def load_artifact(self):
         Returns the nearest cached artifact for this provenance, if one exists.
         """
 
+        try:
+            entry = self._get_nearest_entry_with_artifact()
+
+            if entry is None:
+                return None
+
+            return entry.artifact
+
+        except InternalCacheStateError as e:
+            self.raise_state_error_with_explanation(e)
+
+    def load_local_artifact(self):
+        """
+        Returns a local artifact if any cached artifact exists.
+
+        If the nearest artifact is not local, it will be copied to a local instance and
+        registered in the cache.
```

Comment:
I think this will upload the artifact to the cloud as well? (if so, mention it in the comment)

Reply:
Yes, good point! I wrote the comment before adding the copy-to-cloud functionality. I'll fix it.